### PR TITLE
Migration to Cats Effect 3.0, FS2 3.0

### LINKED ISSUE #946: Support cats-effect 3.0 series and http4s 1.0 series in modules
`cats-effect` 3 has reached RC2 state. It would be nice to start building support for this new library. This will also involve bumping `http4s` to 1.0 series, which supports `cats-effect` 3.0 series.

Both modules are pretty compact, so, it should be relatively easy to implement. The only significant change will be to undeprecate methods w/o `Blocker` instances (because `Blocker` is gone), so, the API of cats-effect module will change in significantly breaking way.

But the problem is that it will be difficult to maintain versions for both series ce3/http4s1 and ce2/http4s0.21 withing the same tree and keep the current versioning scheme. May be it is worth to extract them to separate project/projects and apply reflect version differences there. But this will require some planning from project maintainers anyway.

### PR DESCRIPTION
**Pull request by arixmkii**
Fixes #946 

At least it compiles and passes tests 🤷 But need to check API changes and decide if something else should be changed as well (like dropping more deprecated methods).

### CODE REVIEW

**Comment by rubiconvvv** [comment]
@kubukoz @ruippeixotog hey guys. please check this out, we want to convert our project to cats 3 as soon as possible.

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (kubukoz) [inline modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: I would recommend `F.blocking(...).ifM(ifTrue = ..., ifFalse = ...)` instead, more explicit

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (kubukoz) [inline modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: if the result of `newOutputStream` is a Closeable, you can probably do `Resource.fromAutoCloseable`

**Review by kubukoz — COMMENTED** [review]
In general it looks good :)

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (arixmkii) [reply modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: Thanks! Didn't think about syntax version of `ifM`. Fits really nice here.

**Comment by kubukoz** [comment]
I'll unsubscribe now but feel free to ping me if you need another round.

File: project/Dependencies.scala
Comment (jcazevedo) [inline project/Dependencies.scala]: This update is blocked by an sbt-coverage issue (scoverage/sbt-scoverage#321). Is this required for this PR?

Comment (ruippeixotog) [inline project/Dependencies.scala]: We can't migrate to 2.12.13 yey because scoverage doesn't support it yet (see https://github.com/scoverage/sbt-scoverage/issues/321). This is also what's making the build fail.

Is it required for any of the modules?

File: modules/http4s/build.sbt
Comment (jcazevedo) [inline modules/http4s/build.sbt]: We don't usually publish modules for milestone versions of upstream dependencies. Can we delay this update for when a stable release of http4s in the 1.0 series comes out?

Comment (ruippeixotog) [inline modules/http4s/build.sbt]: How close are we to a RC version here? I'd rather avoid milestone versions if possible.

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
Thanks for the PR @arixmkii! I don't have much experience with cats-effect but I see we already got some eyes on it, so LGTM. I only left a few comments below.

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (jcazevedo) [inline modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: We should remove the deprecated `saveConfigAsPropertyFileF` method now, since it has the same signature as this and its implementation doesn't correctly scope blocking operations.

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (jcazevedo) [inline modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: We should remove the deprecated `saveConfigToStreamF` method now, since it has the same signature as this and its implementation doesn't correctly scope blocking operations.

**Review by jcazevedo — CHANGES_REQUESTED** [review]
@arixmkii: Thanks for this!

Since Cats Effect 3 removes the `Blocker` construct, I don't think it makes much sense to now have both `save...F` and `blockingSave...F` methods. I suggest we move the `blockingSave...F` implementations to the `save...F` ones and deprecate the `blockingSave...F` methods, while removing the deprecation from the `save...F` methods.

File: project/Dependencies.scala
Comment (arixmkii) [reply project/Dependencies.scala]: No, it is not required. I will downgrade.

File: modules/http4s/build.sbt
Comment (arixmkii) [reply modules/http4s/build.sbt]: It is the first version compatible with cats-effect 3.0. Can't tell how close it is to RC state, because there are more changes than just cats-effect 3.0. Parts related to dependencies looks mostly complete for me, but it will need some time to publish with RC tag.

I checked the source of the module. It is not using any effect related methods. It could be possible to compile module with older version, but it will make it impossible to use http4s and cats-effect module in the same application anyway (because of dependency on incompatible versions of cats-effect). If this is ok (at least temporary), it could be downgraded.

File: modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala
Comment (arixmkii) [reply modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala]: Removed deprecated methods and reused their names for blocking implementations.

File: modules/http4s/build.sbt
Comment (arixmkii) [reply modules/http4s/build.sbt]: Downgraded version. Please see my response in different conversation for what it means for compatibility between modules.

File: project/Dependencies.scala
Comment (arixmkii) [reply project/Dependencies.scala]: Downgraded. Bumped it while was looking through code trying to understand the project. Didn't know that there are issues with coverage as used coverage only for Scala 2.13 before.

**Review by ruippeixotog — APPROVED** [review]
LGTM now, thanks. Will let @jcazevedo take a second look and merge this.

**Review by jcazevedo — APPROVED** [review]
@arixmkii: Looks good. Thanks!

**Pull request merged**
